### DISCUSSION TITLE
Create standalone repositories with Quarto book and website reprexes

### DISCUSSION THREAD

**Discussion by andrew-farrey** [quarto-dev/quarto-cli#8751] [Feature Requests]
### Description

Respectfully, I would like to suggest:

- Creating separate github repositories containing reprexes of a Quarto website, book, and other types of output so that people can fork them and have a fresh repository they can use to collaborate from. The built-in Quarto book reprex in the RStudio IDE is awesome, but one shouldn't need to be as gifted at R as Hadley to be able to collaborate with others on a Quarto book using version control. Separate repositories with minimal Quarto reprexes would be easy ways to help make Quarto more broadly accessible.

**Comment by mcanouil** [comment]
Why not use Quarto CLI?
I don't get the possible added value to having to maintain a repository while Quarto CLI has the ability to create projects.

```sh
quarto create project
```

Did you read Quarto documentation?  https://quarto.org/docs/books/


> The built-in Quarto book reprex in the RStudio IDE is awesome

IDEs such as RStudio and VSCode simply make use of Quarto CLI, especially the command shared above.

> but one shouldn't need to be as gifted at R as Hadley to be able to collaborate with others on a Quarto book using version control. 

You are mixing up things.
Quarto is not tied to R.
Quarto has nothing to do with version control.
The user decides to use version control such as Git, decides the Cloud provider such as GitHub and GitLab, decides to use R within his/her project or any language via Jupyter engines.

> Separate repositories with minimal Quarto reprexes would be easy ways to help make Quarto more broadly accessible.

How making repositories would make things more accessible than the command to create project shared above exactly? (I might be missing your rational here)

**Comment by baptiste** [comment]
A few github template repositories would be nice, I agree, as an alternative way to get started (already set up with git by construction, and potentially a bit more fleshed out than the bare-bones skeleton of `quarto create`). 
One issue is that there's a lot of customisation possible, and therefore one template tailored for, say, R users writing a book with a custom TeX template, will not be useful (in fact, confusing) for Python users who want to use Typst for typesetting. The combinatorial explosion of parameters makes it a bit tricky. But perhaps a few generic ones would be useful, if just to use as reprex when posting an issue here? (I've often needed to create a dummy book project to diagnose issues I was having with the real book project, and asking for help is hard when the project involves many files).

**Reply by cscheid** [reply]
> I've often needed to create a dummy book project

In these cases, did the output generated by `quarto create` not serve your needs? That's where we should focus our efforts.

**Reply by mcanouil** [reply]
How `quarto create project` is not enough exactly?
If you know how to fork and clone repositories, you should be able to push a newly created project to any cloud provider such as GitHub.

All the reprexes I made are done this way, see <https://github.com/mcanouil/quarto-issues-experiments>.

**Reply by baptiste** [reply]
Books can require quite a lot of extra customisation, from Latex style files, latex macros, bib files, external figures, css, js, etc. and copying over all these settings to diagnose an issue can take quite a while. There's no easy solution though, unfortunately, as everyone's setup will be quite different.

Another approach I've contemplated was to have a tool create a dummy version of an existing quarto project, where the content files (qmds) are replaced with just a couple of dummy ones, but all the rest of the project's structure is copied over. That would let one post a reprex online without having to worry about sharing sensitive material of the real project. Again not a full solution either, as who can decide which files are structure vs content...

**Reply by baptiste** [reply]
I mean, I can certainly `quarto create` and push things to a repo, and I do it all the time, but I can see that some users less familiar with the command line will find it a barrier. Pushing a button on github to create something from a template is certainly a lower barrier of entry for some, which I think is the OP's point.

**Reply by cscheid** [reply]
> There's no easy solution though, unfortunately, as everyone's setup will be quite different.

> Another approach I've contemplated was to have a tool create a dummy version of an existing quarto project, where the content files (qmds) are replaced with just a couple of dummy ones, but all the rest of the project's structure is copied over.

Right.

I actually think users get a lot of value out of "new project wizards" from IDEs like RStudio, precisely because of the customizability that's possible. The complication arises in how to synchronize what those wizards know how to configure with what Quarto can do independently of the IDE. I can imagine a version of `quarto create` that takes in a YAML file and then configures the "correct kind" of new project, but that takes real developer time and, very plainly, we have more urgent and important things to do than to ship the world's best new project creator.

**Reply by baptiste** [reply]
Perhaps the answer is as easy as having the `quarto create` options also available as github template repos (and obviously updated automatically to keep them in sync with the cli). It's just an alternative interface to the same functionality, for those who prefer to start everything from github's website.

(Edit: I was responding to earlier comments / the OP's suggestion)

**Reply by cscheid** [reply]
> Perhaps the answer is as easy as having the quarto create options also available as github template repos (and obviously updated automatically to keep them in sync with the cli).

They're not different repos, though, they're options that cause us to generate content differently.

**Reply by baptiste** [reply]
I meant that when I run `quarto create project` on the cli, I end up with a website/blog/book/etc. skeleton as a new folder. This skeleton folder could be offered directly as a [github template repository](https://github.com/baptiste/quarto-book), for others to fork on github.com. So there would be a template repo for book, another template repo for website, etc.

<img width="627" alt="Screenshot 2024-02-16 at 16 59 16" src="https://github.com/quarto-dev/quarto-cli/assets/18970/b16deb44-23ec-41c8-8d77-09fbc4f5c0ce">

**Comment by andrew-farrey** [comment]
> I mean, I can certainly `quarto create` and push things to a repo, and I do it all the time, but I can see that some users less familiar with the command line will find it a barrier. Pushing a button on Github to create something from a template is certainly a lower barrier of entry for some, which I think is the OP's point.

@baptiste This, precisely. Offering these skeleton folders as Github template repositories is what I was trying to suggest (and I apologize for the incorrect terminology). Offering Github template repositories would help lower the barrier of entry for R users who aren't accustomed to using the command line and would increase the accessibility of Quarto for people familiar with a specific version control workflow that is frequently taught to R users (I realize Quarto isn't specific to R, but R users are part of the Quarto user base).

Offering template repositories would also align Quarto's offerings more closely with the offerings of the more prolific RMarkdown-focused R packages, which could help drive dedicated users of RMarkdown toward using Quarto. It was just a suggestion though, so it's of course perfectly fine if you all would rather not and/or don't agree with the potential benefit.

Regardless, I appreciate you all taking the time to respond and appreciate the discussion/feedback.
